extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java

```diff
@@ -31,6 +32,15 @@
 
     // Operations
 
+    /**
+     * Returns the default {@link EntityManager} for extra operations (eg. CriteriaQueries)
+     *
+     * @return the default {@link EntityManager}
+     */
+    protected EntityManager entityManager() {
```

Comment:
I think this got merged a bit quickly. Please wait a bit longer to give people time to review.

Why is this protected, and why is this not named `getEntityManager`? In general I don't have anything against dropping `get` from accessor methods but this is not coherent here because we have `isPersistent` in the same class.

Reply:
Thanks for your feedback. May I ask why do you think it should be public? Shouldn't an EntityManager be used only inside the entity? About the method name, I reused what @emmanuelbernard suggested in #7233

Reply:
I believe @emmanuelbernard was in the wrong :)

As for why it should be public, it's forwarding to a public method anyway, so there's no need to hide it, it can always be useful to some people I guess.

Reply:
I didn't notice the `protected` modifier when I review the PR my bad.

I too think it should be public, the main idea behind providing these methods is to allow people to have an easy access to the entity manager (we can already access it via injection as it's a CDI bean, or via `Panache.getEntityManager`) for more complex use cases. 
It's a developer choice to use it only inside the entitry/repository of from a business service layer for example, we should'nt restrict what is possible.

As for the method names, in MongoDB we provides `mongoCollection()` and `mongoDatabase()` to access the underlying low level access objects so it's more natural to provides `entityManager()` for Hibernate Panache. I remember that we discuss to use `get` prefix and there is two reason why I think it's better without it: 
 
1. For entity, `get` is for a getter on the JavaBean standard, so it tells people that it is a field of the object. We can tweak HIbernate to not treat it as a field but it's not natural.
2. The Panache build infrastructure didn't works OOTB with methods starting with `get` as it thinks it's a field getter so it didn't want to inject inside it the implementation (something inside the enhancer ...).

Reply:
@loicmathieu those are really good reasons, and I think I like it without the get prefix too. Please feel free to revert my commit from https://github.com/quarkusio/quarkus/pull/10611 (but making it public) if you manage to convince @FroMage :wink:

Reply:
This has already been merged so let's keep it like this, after all @FroMage is the master of Panache ;)

Reply:
yes I think getEM would likely create more trouble due to the JPA bean property discovery mechanism. Though to be fair I did not think of that when I proposed the name :)